src/app/fdctl/run/tiles/fd_repair.c

```diff
@@ -0,0 +1,656 @@
+handle_new_stake_weights( fd_repair_tile_ctx_t * ctx ) {
+  ulong stakes_cnt = ctx->stake_ci->scratch->staked_cnt;
+
+  if( stakes_cnt >= MAX_REPAIR_PEERS ) {
+    FD_LOG_ERR(( "Cluster nodes had %lu stake weights, which was more than the max of %lu", stakes_cnt, MAX_REPAIR_PEERS ));
+  }
+
+  fd_stake_weight_t const * in_stake_weights = ctx->stake_ci->stake_weight;
+  fd_repair_set_stake_weights( ctx->repair, in_stake_weights, stakes_cnt );
+}
+
+
+static void 
+repair_send_packet( uchar const *                 msg, 
+                    size_t                        msglen, 
+                    fd_gossip_peer_addr_t const * addr, 
+                    void * arg ) {
+  ulong tsorig = fd_frag_meta_ts_comp( fd_tickcount() );
+  send_packet( arg, addr->addr, fd_ushort_bswap( addr->port ), msg, msglen, tsorig );
+}
+
+static void
+repair_shred_deliver( fd_shred_t const *            shred,
+                      ulong                         shred_sz, 
+                      fd_repair_peer_addr_t const * from FD_PARAM_UNUSED, 
+                      fd_pubkey_t const *           id FD_PARAM_UNUSED, 
+                      void *                        arg ) {
+  fd_repair_tile_ctx_t * ctx = (fd_repair_tile_ctx_t *)arg;
+  
+  fd_shred_t * out_shred = fd_chunk_to_laddr( ctx->store_out_mem, ctx->store_out_chunk );
+  fd_memcpy( out_shred, shred, shred_sz );
+  
+  ulong tspub = fd_frag_meta_ts_comp( fd_tickcount() );
+  ulong sig = 0UL;
+  fd_mux_publish( ctx->mux, sig, ctx->store_out_chunk, shred_sz, 0UL, 0UL, tspub );
+  ctx->store_out_chunk = fd_dcache_compact_next( ctx->store_out_chunk, shred_sz, ctx->store_out_chunk0, ctx->store_out_wmark );}
+
+static void
+repair_shred_deliver_fail( fd_pubkey_t const * id FD_PARAM_UNUSED, 
+                           ulong               slot, 
+                           uint                shred_index,
+                           void *              arg FD_PARAM_UNUSED,
+                           int                 reason ) {
+  FD_LOG_WARNING(( "repair failed to get shred - slot: %lu, shred_index: %u, reason: %u", slot, shred_index, reason ));
+}
+
+static void
+before_frag( void * _ctx FD_PARAM_UNUSED,
+             ulong  in_idx,
+             ulong  seq  FD_PARAM_UNUSED,
+             ulong  sig,
+             int *  opt_filter ) {
+
+  if( FD_LIKELY( in_idx==NET_IN_IDX ) ) {
+    *opt_filter = fd_disco_netmux_sig_proto( sig )!=DST_PROTO_REPAIR;
+  }
+}
+
+static void
+during_frag( void * _ctx,
+             ulong  in_idx,
+             ulong  seq FD_PARAM_UNUSED,
+             ulong  sig,
+             ulong  chunk,
+             ulong  sz,
+             int *  opt_filter ) {
+
+  fd_repair_tile_ctx_t * ctx = (fd_repair_tile_ctx_t *)_ctx;
+  uchar const * dcache_entry;
+  if( FD_UNLIKELY( in_idx==CONTACT_IN_IDX ) ) {
+    if( FD_UNLIKELY( chunk<ctx->contact_in_chunk0 || chunk>ctx->contact_in_wmark ) ) {
+      FD_LOG_WARNING(( "chunk %lu %lu corrupt, not in range [%lu,%lu]", chunk, sz,
```

Comment:
These should all log error and abort. No need to filter the frag. Internal packet corruption is a critical error.